lillisp.go

```diff
@@ -3,15 +3,78 @@ package main
 import (
 	"bufio"
 	"fmt"
+	"log"
 	"os"
+	"strconv"
 	"unicode/utf8"
 )
 
+// Atom ...
+type Atom interface{}
```

Comment:
maybe a different name for this type like LispObject or even Thing would be better since atoms are a realized type I was just using string for atom but we could make a real type for it